extensions/amp-social-share/0.1/test/test-amp-social-share.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {adopt} from '../../../../src/runtime';
+import {createIframePromise} from '../../../../testing/iframe';
+import '../../../../build/all/v0/amp-social-share-0.1.max';
+
+adopt(window);
+
+describe('amp-social-share', () => {
+
+  function getShare(type, config) {
+    return createIframePromise().then(iframe => {
+      const share = iframe.doc.createElement('amp-social-share');
+      const script = iframe.doc.createElement('script');
+
+      script.setAttribute("type", "application/json");
+      script.textContent = JSON.stringify(config);;
+
+      share.setAttribute("type", type);
+      share.setAttribute('width', 60);
+      share.setAttribute('height', 44);
+      share.appendChild(script);
+      return iframe.addElement(share);
+    });
+  }
+
+
+  it('renders twitter', () => {
```

Comment:
Need more tests. Mostly this is important for different DOM rendering cases. But also other nuances important, such as `maxLength` restrictions, etc.